rust/kernel/sync/arc.rs

```diff
@@ -610,24 +610,24 @@ impl<T: ?Sized> DerefMut for UniqueArc<T> {
 
 impl<T: fmt::Display + ?Sized> fmt::Display for UniqueArc<T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        fmt::Display::fmt(self.deref(), f)
+        fmt::Display::fmt(&self, f)
```

Comment:
Hmm.. I don't think this is an improve, `&self` is of type `&&UniqueArc<T>`, it's not very straight-forwards why does the compiler coerce it to a `&T`. We could do the same as std:
```rust
fmt::Display::fmt(&**self, f);
```
But honestly, I don't think it's better.. could you explain what problem the old code has?

Reply:
There is no particular problem other than keeping consistency, this comes from the [`explicit_deref_methods`](https://rust-lang.github.io/rust-clippy/master/index.html#/explicit_deref_methods) lint. I suspect that this is why `std` does what it does (I neglected to check what they do before applying this).

I am indifferent about this lint and agree that there are many times where it makes more sense to be explicit, so I don't mind dropping it. However, even unrelated to the lint, maybe it would be good to match what `std` does, just to avoid any possible confusion when syncing `std` -> `kernel`